indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -83,6 +92,47 @@ protected void map(
     }
   }
 
-  abstract protected void innerMap(InputRow inputRow, Text text, Context context)
+  public final static InputRow parseInputRow(Writable value, InputRowParser parser)
+  {
+    if(value instanceof Text) {
+      //Note: this hack is required because StringInputRowParser is a
+      //InputRowParser<ByteBuffer> and not InputRowParser<String>
+      //parse() method on it expects a ByteBuffer and not String
+      ByteBuffer bb = ByteBuffer.wrap(value.toString().getBytes());
```

Comment:
StringInputRowParser expects UTF-8. Unless it is guaranteed in the spec somehow that text writeable's bytes are UTF-8 encoded, I would prefer we do it this way but convert to UTF-8 so that we are absolutely sure we're passing text formats correctly.